### ISSUE TITLE
increment_build_number does not update extension

### ISSUE THREAD

**Issue by phlippieb** [fastlane/fastlane#15093]
### Question Checklist

- [✅] Updated fastlane to the latest version
- [✅] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [✅] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [✅] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Question Subject

increment_build_number, when project has extension targets

#### Question Description

We have an iOS project with multiple targets. (We have used fastlane with this project.) In our fastfile, we a lane to build and distribute the ad-hoc target. Here's a relevant snippet from the fastfile:

```
lane :beta_gym do
        ensure_git_status_clean

        match(
            git_url: "...", # redacted
            type: "adhoc",
            app_identifier: "...", # redacted
            verbose: false,
            readonly: true,
            clone_branch_directly: true)

        increment_build_number

        changelog = changelog_from_git_commits(
            pretty: '- (%ae) %s',
            include_merges: false)
        
        gym(
            silent: true,
            verbose: false,
            scheme: "STAGING")
```

The `increment_build_number` action has always worked perfectly across all targets. But recently, we've added new app extension targets to allow users to share content into our app. We created these extensions targets by embedding them in the existing targets, creating common app groups, etc. Now, when fastlane runs `increment_build_number`, the main app targets still get their build numbers incremented, BUT the build numbers of the extension targets remain unchanged.

This is a serious issue for us. We use Realm as our in-app database, and Realm uses the app versions to determine whether to migrate data. Realm does not allow migrating backwards. So if the main app and the extension each configure Realm with a different app version, then the extension (which will have a lower app version than the app) is not able to open Realm, and will always crash when opened.

Is there a hidden setting that we should pass to `increment_build_number` in order for it to also affect the extension targets? Or is there a way to make the extension inherit its version from the main app it is embedded in that we don't know about?

**Comment by phlippieb** [comment]
Aha

If I just run `fastlane run increment_build_number`, I get the following errors:

```
[14:46:12]: Cannot set build number with plist path containing $(SRCROOT)
[14:46:12]: Please remove $(SRCROOT) in your Xcode target build settings
```

So I navigated to `My project (blue icon)` -> `extension target` -> `Build settings`, and searched for "info.plist file". When expanding the plist file setting, I could see that (for the extension targets) the path included a `$(SRCROOT)` variable. It evaluated to the correct path, but apparently this is an issue for agvtool. So I deleted the variable. I.e I changed it from

```
$(SRCROOT)/Builds/Products/extension_target/Info.plist
```
to
```
Builds/Products/extension_target/Info.plist
```

After I did this for each extension target, I could run increment_build_number successfully.
